pandas/core/frame.py

```diff
@@ -217,11 +217,16 @@ def __init__(self, data=None, index=None, columns=None, dtype=None,
         if isinstance(data, DataFrame):
             data = data._data
 
+        if hasattr(data, 'to_dataframe'):  # xr.Dataset
```

Comment:
I'm happy with a duck-like method, but we should probably call it something very explicit like  `_to_pandas_dataframe_`. This would be useful for a lot of other projects, not just xarray.